### PR TITLE
Added support for GenomeReference build38

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: src/main/resources/reference/human_g1k_v37.json [hunk 1]
```diff
@@ -1,393 +1 @@
...
...h": 161147},{"name": "GL000220.1","length": 161802},{"name": "GL000213.1","length": 164239},{"name": "GL000211.1","length": 166566},{"name": "GL000199.1","length": 169874},{"name": "GL000217.1","length": 172149},{"name": "GL000216.1","length": 172294},{"name": "GL000215.1","length": 172545},{"name": "GL000205.1","length": 174588},{"name": "GL000219.1","length": 179198},{"name": "GL000224.1","length": 179693},{"name": "GL000223.1","length": 180455},{"name": "GL000195.1","length": 182896},{"name": "GL000212.1","length": 186858},{"name": "GL000222.1","length": 186861},{"name": "GL000200.1","length": 187035},{"name": "GL000193.1","length": 189789},{"name": "GL000194.1","length": 191469},{"name": "GL000225.1","length": 211173},{"name": "GL000192.1","length": 547496}],"xContigs": ["X"],"yContigs": ["Y"],"mtContigs": ["MT"],"par": [{"start": {"contig": "X","position": 60001},"end": {"contig": "X","position": 2699521}},{"start": {"contig": "X","position": 154931044},"end": {"contig": "X","position": 155260561}},{"start": {"contig": "Y","position": 10001},"end": {"contig": "Y","position": 2649521}},{"start": {"contig": "Y","position": 59034050},"end": {"contig": "Y","position": 59363567}}]}
```
Comment (cseed) [inline src/main/resources/reference/human_g1k_v37.json]: What's this address, `ftp.broadinstitute.org/bundle/b37/human_g1k_v37.dict`?  I don't see any directory `bundle` there.

File: src/main/scala/is/hail/variant/GenomeReference.scala [hunk 2]
```diff
@@ -10,24 +11,95 @@ import org.json4s.jackson.JsonMethods
...
+      mtContigs == other.mtContigs &&
+      par.sameElements(other.par)
+  }
 }
 
 object GenomeReference {
+
   def GRCh37 = fromResource("reference/human_g1k_v37.json")
 
+  def GRCh38 = fromResource("reference/Homo_sapiens_assembly38.json")
```
Comment (cseed) [inline src/main/scala/is/hail/variant/GenomeReference.scala]: Do we want more consistent names?  Is there any point?  What's `g1k`?

File: src/main/resources/reference/human_g1k_v37.json (hunk 1 above)
Comment (jigold) [inline src/main/resources/reference/human_g1k_v37.json]: That's where I had downloaded the original file with the data. I updated the URL. Should have been: ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/

File: src/main/scala/is/hail/variant/GenomeReference.scala (hunk 2 above)
Comment (jigold) [inline src/main/scala/is/hail/variant/GenomeReference.scala]: I thought it was better to keep the name of the source file. However, that was before I put the source into the JSON. Will change the name.

**Pull request merged**